frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx

```diff
@@ -58,42 +58,116 @@ import {
 import { FieldGroup } from './descriptors/spec/field-group';
 import { referenceForProvidedAPI, ClusterServiceVersionLogo, providedAPIsFor } from './index';
 
-const annotationKey = 'alm-examples';
+/*
+ * Matches a path that contains an array index. Use Sting.match against an OperandField 'path'
+ * property to determine if it contains an array index. It will parse the path into three parts,
+ * [match, pathBeforeIndex, index, pathAfterIndex]. For example:
+ *
+ *   const [match, pathBeforeIndex, index, pathAfterIndex] =
+ *     'path.before[0].path.after'.match(ARRAY_INDEX_PATTERN);
+ *
+ *   console.log(match);
+ *   > 'path.before[0].path.after'
+ *
+ *   console.log(pathBeforeIndex);
+ *   > 'path.before'
+ *
+ *   console.log(index)
+ *   > '0'
+ *
+ *   console.log(pathAfterIndex)
+ *   > 'path.after'
+ *
+ */
+const ARRAY_INDEX_PATTERN = /^(.*)\[(\d)\]\.?(.*)$/;
+
+// Regex for SpecCapability.arrayFieldGroup and SpecCapability.fieldGroup
+const ARRAY_FIELD_GROUP_PATTERN = _.escapeRegExp(SpecCapability.arrayFieldGroup);
+const FIELD_GROUP_PATTERN = _.escapeRegExp(SpecCapability.fieldGroup);
+
+/*
+ * Matches either field group or array field group SpecCapabilty. Use String.match against an
+ * OperandField.capability element to determine if it is a group descriptor and parse group type
+ * and name. For example:
+ *
+ *   const [match, groupType, name] = 'urn:alm:descriptor:com.tectonic.ui:fieldGroup:groupName';
+ *
+ *   console.log(match);
+ *   > 'urn:alm:descriptor:com.tectonic.ui:fieldGroup:groupName'
+ *
+ *   console.log(groupType);
+ *   > 'urn:alm:descriptor:com.tectonic.ui:fieldGroup:'
+ *
+ *   console.log(name);
+ *   > 'groupName'
+ */
+const GROUP_PATTERN = new RegExp(`^(${FIELD_GROUP_PATTERN}|${ARRAY_FIELD_GROUP_PATTERN})(.*)$`);
 
-enum Validations {
-  maximum = 'maximum',
-  minimum = 'minimum',
-  maxLength = 'maxLength',
-  minLength = 'minLength',
-  pattern = 'pattern',
-}
+// Default max nesting depth the form should display
+const MAX_DEPTH = 1;
```

Comment:
Added constants to hold regex patterns for parsing spec descriptors and paths, as well as a default max depth which represents how deeply nested the form can be. The form implementation is still not quite ready for this number to be arbitrary though.

Reply:
And defaulting to 1?

Reply:
Yeah, that is the current behaviour. We show one level of nesting.